### PR TITLE
Quick Start for Existing Users V2 - Refactor code to support different quick start types (FluxC)

### PR DESCRIPTION
**Pull request by ashiagr**
Parent https://github.com/wordpress-mobile/WordPress-Android/issues/16350

This PR splits `QuickStartTask` into two different sets of tasks for different quick start types. 
PS: Currently, no new tasks are added to the "existing site"  quick start type.

## How to test?

See corresponding WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/16500

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/QuickStartStore.kt [hunk 1]
```diff
@@ -4,51 +4,96 @@ import org.greenrobot.eventbus.Subscribe
...
+            fun fromString(string: String?): QuickStartExistingSiteTask {
+                for (value in values()) {
+                    if (string.equals(value.toString(), true)) {
+                        return value
+                    }
+                }
+
+                return UNKNOWN
             }
         }
```
Comment (AjeshRPai) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/QuickStartStore.kt:98]: Nitpick:  @ashiagr I think the following functions are redundant and can be removed,  Correct me if am wrong 
1. `QuickStartTaskType.fromString`
2. `getShownCount`
3. `hasShownTask`
4. `setShownTask `
WDYT? 🤔

**Review by AjeshRPai — APPROVED** [review]
I have looked at the code changes and tested it as per the instructions. Everything looks good to me. Great Job @ashiagr. 👍 🙌 . 

I left a nitpick, please feel free to merge once you have addressed it. Approving the PR as the Nitpick is non blocking.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/QuickStartStore.kt (hunk 1 above)
Comment (ashiagr) [reply fluxc/src/main/java/org/wordpress/android/fluxc/store/QuickStartStore.kt:98]: I removed `QuickStartTaskType.fromString` in 80d5012.  We can remove pre-existing unused functions related to "task shown" in a separate cleanup PR as it impacts SQL util functions and tests cases as well which are not related to changes in this PR.

I'll shortly create one.

Comment (ashiagr) [reply fluxc/src/main/java/org/wordpress/android/fluxc/store/QuickStartStore.kt:98]: Never mind, I removed them in 2a66831.

**Pull request merged**